packages/web3-errors/src/web3_error_base.ts

```diff
@@ -50,8 +52,24 @@ export abstract class Web3Error extends Error implements ErrorInterface {
 			: result;
 	}
 
-	public toJSON() {
-		return { name: this.name, code: this.code, message: this.message };
+	public toJSON(): {
+		name: string;
+		code: number;
+		message: string;
+		innerError?: Error;
+	} {
+		if (this.innerError)
+			return {
+				name: this.name,
+				code: this.code,
+				message: this.message,
+				innerError: this.innerError,
+			};
+		return {
+			name: this.name,
+			code: this.code,
+			message: this.message,
+		};
```

Comment:
if innerError is undefined it will be same in json. 
```suggestion
			return {
				name: this.name,
				code: this.code,
				message: this.message,
				innerError: this.innerError,
			};
```